build-logic/jvm/src/main/kotlin/gradlebuild/jvm/extension/UnitTestAndCompileExtension.kt

```diff
@@ -50,8 +54,11 @@ abstract class UnitTestAndCompileExtension(private val tasks: TaskContainer) {
     fun enforceJava6Compatibility() {
         tasks.withType<JavaCompile>().configureEach {
             options.release.set(null)
+            options.compilerArgs.remove("-parameters")
             sourceCompatibility = "6"
             targetCompatibility = "6"
         }
+        // Apply ParameterNamesIndex since 6 doesn't support -parameters
+        project.apply(plugin = "gradlebuild.api-parameter-names-index")
```

Comment:
are there a small enough number of projects that we could apply this to them specifically? it's a little ugly to apply a plugin from a method call like this.

Reply:
There are 23 projects, it's do-able but maybe a little much, I would rather keep this consistent as it should be applied to any project that has Java 6 applied.